lib/LaTeXML/Package/LaTeX.pool.ltxml

```diff
@@ -1689,37 +1704,37 @@ DefPrimitiveI('\@vobeyspaces', undef, sub {
 # WARNING: Need to be careful about what catcodes are active here
 DefMacroI('\verb', undef, sub {
     my ($gullet) = @_;
-    my ($init, $body);
+    my $inmath = LookupValue('IN_MATH');
     StartSemiverbatim('%', '\\', '{', '}');
-    $init = $gullet->readToken;
-    $init = $gullet->readToken if ToString($init) eq '*';    # Should I bother handling \verb* ?
-
+    MergeFont(family => 'typewriter', series => 'medium', shape => 'upright');
+    $STATE->assignCatcode(' ', CC_ACTIVE);    # Do NOT (necessarily) skip spaces after \verb!!!
+    Let(T_ACTIVE(' '), T_SPACE);
+    my $init = $gullet->readToken;
+    if (T_OTHER('*')->equals($init)) {
+      Let(T_ACTIVE(' '), T_OTHER("\x{2423}"));    # Visible space
+      $init = $gullet->readToken; }
     if (!$init) {    # typically something read too far got \verb and the content is somewhere else..?
       Error('expected', 'delimiter', $gullet,
         "Verbatim argument lost", "Bindings for preceding code is probably broken");
       EndSemiverbatim();
       return (); }
-    my $old_cc   = $$init[1];
     my $init_str = $init->getString();
     AssignCatcode($init_str => CC_ACTIVE);
-    $body = $gullet->readUntil(T_ACTIVE($init_str));
-    AssignCatcode($init_str => $old_cc);
-    EndSemiverbatim();
-    Invocation((LookupValue('IN_MATH') ? T_CS('\@math@verb') : T_CS('\@text@verb')),
+    my $delim = T_ACTIVE($init_str);
+    my $body  = $gullet->readUntil($delim);
+    Invocation(($inmath ? T_CS('\@math@verb') : T_CS('\@text@verb')),
       Tokens($init), $body)->unlist; });
 
 DefConstructor('\@text@verb{}{}', "<ltx:verbatim font='#font'>#2</ltx:verbatim>",
-  beforeDigest => [sub { $_[0]->bgroup; MergeFont(family => 'typewriter'); }],
-  afterDigest  => sub { $_[0]->egroup; },
+  afterDigest => sub { EndSemiverbatim(); },
   # Since ltx:verbatim is both inline & block, we have to fudge inline mode
   beforeConstruct => sub {
     $_[0]->canContain($_[0]->getElement, '#PCDATA')
       || $_[0]->openElement('ltx:p'); },
   reversion => '\verb#1#2#1');
```

Comment:
while reviewing here I just noticed that the `\verb*` variant will also be reverted into `\verb`. Not critical for anything today, but noting in case you feel like patching.

Reply:
Good catch; better fix now, else it'll show up later.